GUIDE.md

```diff
@@ -0,0 +1,1021 @@
+
+If you do something smart and useful as a `project->action` you can easily share this to be plugged into other `web-app-react` projects by moving it into a [dedicated roc plugin](https://github.com/rocjs/roc/blob/master/docs/guides/CreateExtension.md#creating-an-extension).
+
+### Babel
+
+Additions can be made to the babel configuration in several ways:
+1. Add a `babel` property to your `roc.config.js`
+2. Add `.babelrc` file to the project
+3. Add a `babel` property to your `package.json`
+
+Configuration is prioritized in that order. Only one of them will be considered.
+
+Note that all Babel configurations will support the `extends` property. The value of this determines if the project Babel configuration should be [merged](https://github.com/schnittstabil/merge-options) with the one provided by `web-app-react` or not. It will extend by default. Set to `false` if you which to overwrite the entire configuration.
+
+#### Using roc.config.js
+
+Example `roc.config.js` with babel object:
+```javascript
+module.exports = {
+  settings: {
+    // your settings...
+  },
+  babel: {
+    extends: false, // we want to define a config from scratch
+    // your custom babel config
+  },
+}
+```
+
+Example `roc.config.js` with babel function:
+```javascript
+module.exports = {
+  settings: {
+    // your settings...
+  },
+  babel: (target) => {
+    return {
+      extends: true, // we want to merge with the one provided by web-app-react
+      // your custom babel config depending on target
+    };
+  },
+}
+```
+
+#### Using .babelrc or package.json
+If you use `.babelrc` or `babel` definition in `package.json` it works just like [documented by Babel](https://babeljs.io/docs/usage/babelrc/).
+
+`extends` set to `false` is still respected if you wish to ignore the values defined by `web-app-react`.
+
+#### With a project action
+
+Like explained in the Webpack section above; a dedicated hook is exposed for modifying Babel configurations. It is named `babel-config`.
+
+Example `roc.config.js` with custom action function to modify Babel config:
+
+```javascript
+module.exports = {
+  settings: {
+    // your settings...
+  },
+  project: {
+    actions: ({ hook }) => {
+      if (hook === 'babel-config') {
+        return () => (babel) => {
+          // do something with babel
+          return babel;
+        }
+      }
+    }
+  },
+}
+```
+
+You can make sharable Babel adjustments between `web-app-react` projects by moving the action into a [dedicated roc plugin](https://github.com/rocjs/roc/blob/master/docs/guides/CreateExtension.md#creating-an-extension).
+
+## Polyfills
+
+`web-app-react` does not add any polyfills for you. If you wish to add custom polyfills this can be done by intalling the required polyfills to your project. For example:
+
+`$ npm i --save core-js`
+
+Add a file that can group your polyfilling ambitions, `src/polyfills.js` to the project:
+
+```javascript
+// only include polyfills on the client
+if (__WEB__) {
```

Comment:
Is this for illustrative purposes or is it something the stack exposes for us?

Reply:
Actually this is available for us in the project and can be used for things like this.

I'll add a section about that, as it is pretty useful 👍

Reply:
Added here https://github.com/rocjs/roc-package-web-app-react/pull/78/commits/9ff220197d8af4e8f2bd96db513fac22768861c4